button-card.js

```diff
@@ -121,118 +127,173 @@
       return retval;
     }
 
-    buildCssColorAttribute(state, config) {
-      let color = config.color;
-      if (state) {
-        let configState = this.testConfigState(state, config);
-        if (configState) {
-          color = configState.color ? configState.color : config.color_off;
-          if (configState.color === 'auto') {
-            color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : configState.default_color;
-          }
+    buildCssColorAttribute(state, config, configState) {
+      let colorValue = null;
+      if (configState && configState.color) {
+        colorValue = configState.color;
+      } else {
+        if (config.color != 'auto' && state && state.state == 'off')
+          colorValue = config.color_off;
+        else
+          colorValue = config.color;
+      }
+      if (colorValue == 'auto') {
+        if (state) {
+          const color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : config.default_color;
+          return color;
         } else {
-          if (config.color === 'auto') {
-            color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : config.default_color;
+          return config.default_color;
+        }
+      } else {
+        if (!colorValue) {
+          if (state) {
+            if (state.state == 'off') {
+              const color = config.color_off;
+              return color;
+            } else {
+              const color = config.default_color;
+              return color;
+            }
+          } else {
+            const color = config.default_color;
+            return color;
           }
-          color = state.state === 'on' ? color : config.color_off;
+        } else {
+          const color = colorValue;
+          return color;
         }
       }
-      return color;
     }
 
-    buildIcon(state, config) {
-      let iconOff = config.icon;
-      if (config.icon == 'attribute') {
+    buildIcon(state, config, configState) {
+      let iconValue = null;
+      if (configState && configState.icon) {
+        iconValue = configState.icon;
+      } else {
+        iconValue = config.icon;
+      }
+      if (iconValue == 'attribute') {
         if (state) {
           const icon = state.attributes.icon;
           return icon;
+        } else {
+          return undefined;
         }
-        return iconOff;
-      }
-      let configState = this.testConfigState(state, config);
-      if (configState && configState.icon) {
-        const icon = configState.icon;
+      } else {
+        const icon = iconValue;
         return icon;
       }
-      return iconOff;
     }
 
-    blankCardColoredHtml(state, config) {
+    buildStyle(state, config, configState) {
+      let cardStyle = '';
+      let styleArray = undefined;
+      if (state) {
+        if (configState && configState.style)
```

Comment:
This code could use some linting love.
```suggestion
        if (configState && configState.style) {
            styleArray = configState.style;
        }
```

I'm not going to be a huge stickler or anything, but for the sake of my eyes I'll throw out some suggestions :)